src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -495,9 +495,15 @@ else if (SiteUtil.JAVADOC_TOKENS.equals(propertyName)) {
             writeTokensList(sink, configurableTokens, SiteUtil.PATH_TO_JAVADOC_TOKEN_TYPES, true);
         }
         else {
-            final String defaultValue = SiteUtil.getDefaultValue(
-                    propertyName, field, instance, currentModuleName);
+            final String defaultValue;
 
+            if (field != null) {
+                defaultValue = SiteUtil.getDefaultValue(
+                        propertyName, field, instance, currentModuleName);
+            }
+            else {
+                defaultValue = CURLY_BRACKET;
```

Comment:
This is not what follows at https://github.com/checkstyle/checkstyle/pull/13964#discussion_r1382603681 . It needs to go against the setters.

Reply:
Can we postpone it  until https://github.com/checkstyle/checkstyle/pull/14048#discussion_r1399817328 ?

Reply:
I understand. If we have this piece of code in `getDefaultValue`, it would look something like this:
public static String getDefaultValue(String propertyName, Field field,
                                         Object classInstance, String moduleName)
            throws MacroExecutionException {
```
 final Object value;
        if (field == null) {
                   // if there is no class field, there is no default value.
                    value = null
              }
       } esle {
        value = getFieldValue(field, classInstance);
}
...
}
```
We neednot assign any value apart from null. Basically, i had to assign `CURLY_BRACKETS`, because the code in getDefaultValue would eventually do that.

Reply:
For this check. You can't say it will be curlies for all. Curly is based on the type of this check. If we have a check with a boolean property, you can't say the default value will be a curly.

Reply:
Agreed, if we move it back to `getDefaultValue` or part of any other improvement, we need not self-assign as curly. I guess, the exisiting code will determine the apt value. However, we could just assign the value as null. However, as Romani suggested, this wasn't a good practice to go inside get `getDefaultValue` if the field is itself `null`.

Reply:
> Romani suggested, this wasn't a good practice to go inside get getDefaultValue if the field is itself null.

He can suggest then how to improve the code. I am just saying this code will not work for all cases and will eventually fail. We should try to fix it now before it fails, because no one will come back to this and eventually forget.

Reply:
Yes, current PR will work only of this module and property, improvement will be must to do in my PR.